### ISSUE TITLE
Check constraints before adding statement

### ISSUE THREAD

**Issue by dpriskorn** [LeMyst/WikibaseIntegrator#154]
To take WBI to the next level I propose to enable optional checking by default of whether the statement violates a constraint using the plnode API.

see https://plnode.toolforge.org/

**Comment by LeMyst** [comment]
That seem interesting. Thank you.
I'll finish the rewrite of WBI before implementing new things.

**Comment by shigapov** [comment]
Indeed interesting! But right now plnode is adapted to Wikidata. AFAIK, the third-party Wikibase instances do not use constraints so far.

**Comment by LeMyst** [comment]
> Indeed interesting! But right now plnode is adapted to Wikidata. AFAIK, the third-party Wikibase instances do not use constraints so far.

Yeah, after a second check and reading the source of the tool, it seem the tool is too Wikidata-oriented.
QualityConstraints can be installed on a Wikibase instance, but this tool has hardcoded Constraints.

I'll keep in mind the constraint check.

**Comment by dpriskorn** [comment]
I removed the default then, as this tool should just work.
Maybe we can have Wikibase read all constraints of the wikibase installation and compile an internal database?
Or improve the module you mentioned with an API?
